docs/modules/ROOT/pages/contributing/developers.adoc

```diff
@@ -22,18 +22,17 @@ There are many ways you can contribute to Camel K, not only software development
 
 In order to build the project, you need to comply with the following requirements:
 
-* **Go version 1.16+**: needed to compile and test the project. Refer to the https://golang.org/[Go website] for the installation.
+* **Go version 1.21+**: needed to compile and test the project. Refer to the https://golang.org/[Go website] for the installation.
```

Comment:
We could try to automate and scan the go.mod to get such info during nightly tasks.